apps/bot_manager/lib/game_socket_handler.ex

```diff
@@ -44,9 +44,29 @@ defmodule BotManager.GameSocketHandler do
       %{event: {:update, game_state}} ->
         bot_player = Map.get(game_state.players, state.player_id)
 
+        state =
+          if Map.has_key?(state, :bot_skills) do
+            state
+          else
+            {:player, aditional_info} = bot_player.aditional_info
+
+            skills =
+              BotManager.Utils.list_character_skills_from_config(aditional_info.character_name, state.config.characters)
+
+            Map.put(state, :bot_skills, skills)
+          end
+
+        bot_state_machine =
```

Comment:
I’m determining that a character is a melee type if their basic skill is melee-based. However, this could be a wrong assumption because characters like Shinko or Uma have abilities that affect an area of effect (AoE) centered around them, even though they might not be strictly melee-focused.